File: src/main/java/edu/harvard/mcz/imagecapture/utility/EventEmitter.java
package edu.harvard.mcz.imagecapture.utility;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.ArrayList;

public abstract class EventEmitter {

    protected ArrayList<ActionListener> listeners;

    public void addListener(ActionListener listener) {
        listeners.add(listener);
    }

    public void removeListener(ActionListener listener) {
        listeners.remove(listener);
    }

    protected void emitEvent(ActionEvent event) {
        for (ActionListener listener :
                listeners) {
            listener.actionPerformed(event);
        }
    }
}

File: src/main/kotlin/edu/harvard/mcz/imagecapture/utility/EventEmitter.kt
package edu.harvard.mcz.imagecapture.utility


import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import java.util.*

abstract class EventEmitter {
    protected var listeners: ArrayList<ActionListener?>? = null
    fun addListener(listener: ActionListener?) {
        listeners!!.add(listener)
    }

    fun removeListener(listener: ActionListener?) {
        listeners!!.remove(listener)
    }

    protected fun emitEvent(event: ActionEvent?) {
        for (listener in listeners!!) {
            listener.actionPerformed(event)
        }
    }
}
